pulsar-client-cpp/lib/ConsumerImpl.cc

```diff
@@ -303,7 +299,6 @@ void ConsumerImpl::unsubscribeAsync(ResultCallback callback) {
 
 void ConsumerImpl::handleUnsubscribe(Result result, ResultCallback callback) {
     if (result == ResultOk) {
-        Lock lock(mutex_);
         state_ = Closed;
```

Comment:
Have a question:

What is the difference between direct assignment and `state_.store()`? Is there a visibility issue with the direct assignment?

Reply:
> Have a question:
> 
> 
> 
> What is the difference between direct assignment and `state_.store()`? Is there a visibility issue with the direct assignment?
> 
> 

The atomic variable overloaded the operator `=`, it is
equivalent to `store(desired)`.
More see：https://en.cppreference.com/w/cpp/atomic/atomic/operator%3D